Use DEBUG2 for ANALYZE debug logging rather than LOG

danielgustafsson opened the pull request:
Running ANALYZE with the HLL computation produce a lot of LOG messages which are more geared towards troubleshooting than general purpose log files. Demote these to elevel DEBUG2 to avoid cluttering up logfiles on production systems.

Also fixes a C++ comment to conform with code style which was in the vicinity

ashwinstar commented on src/test/regress/expected/incremental_analyze.out:
```diff
@@ -1690,10 +1690,6 @@ SET client_min_messages = 'log';
 ALTER TABLE foo ADD COLUMN c int;
 INSERT INTO foo SELECT i, i%9, i%100 FROM generate_series(1,500)i;
 ANALYZE rootpartition foo;
-LOG:  column c of partition foo_1_prt_1 is not analyzed, so ANALYZE will collect sample for stats calculation
```
@danielgustafsson wish to get this fixed and in? Seeing many logs dumped during initdb is annoying.

danielgustafsson replied:
I have it more or less fixed, rebased and ready to go, will try to wrap it up shortly once I'm not giving training.

danielgustafsson commented:
I've pushed an update which addresses the review comment. I'm not really thrilled about the implementation, and the way we need to punch a hole into the deeper layers via `VacAttrStats`, but I can't see a better way off the cuff.

danielgustafsson commented:
> Ok. It's a bit sad to have to pass `elevel` through functions like this, but I don't see any better way.

Yeah .. if we figure out a better way down the line we can always fix it up but for now it seems the least bad option. Thanks for review!

danielgustafsson commented:
Pushed to master
